GVRf/Framework/src/org/gearvrf/scene_objects/GVRTextViewSceneObject.java

```diff
@@ -0,0 +1,310 @@
+            int viewHeight, String text) {
+        this(gvrContext, gvrContext.createQuad(width, height), gvrActivity,
+                viewWidth, viewHeight, text);
+    }
+
+    /**
+     * Shows a {@link TextView} on a {@linkplain GVRSceneObject scene object}
+     * with view default height and width.
+     * 
+     * @param gvrContext
+     *            current {@link GVRContext}
+     * @param mesh
+     *            a {@link GVRMesh} - see
+     *            {@link GVRContext#loadMesh(org.gearvrf.GVRAndroidResource)}
+     *            and {@link GVRContext#createQuad(float, float)}
+     * @param gvrActivity
+     *            a {@link GVRActivity}
+     * @param text
+     *            {@link String} to show on the textView
+     */
+
+    public GVRTextViewSceneObject(GVRContext gvrContext, GVRMesh mesh,
+            GVRActivity gvrActivity, String text) {
+        this(gvrContext, mesh, gvrActivity, DEFAULT_WIDTH, DEFAULT_HEIGHT, text);
+    }
+
+    /**
+     * Shows a {@link TextView} on a {@linkplain GVRSceneObject scene object}
+     * with view's default height and width.
+     * 
+     * @param gvrContext
+     *            current {@link GVRContext}
+     * @param gvrActivity
+     *            a {@link GVRActivity}
+     * @param width
+     *            width for createQuad to get a new {@link GVRMesh}
+     * @param height
+     *            height for createQuad to get a new {@link GVRMesh}
+     * @param text
+     *            {@link String} to show on the textView
+     */
+
+    public GVRTextViewSceneObject(GVRContext gvrContext,
+            GVRActivity gvrActivity, float width, float height, String text) {
+        this(gvrContext, gvrContext.createQuad(width, height), gvrActivity,
+                text);
+    }
+
+    /**
+     * Shows a {@link TextView} on a {@linkplain GVRSceneObject scene object}
+     * with both view's default height and width and quad's default height and
+     * width.
+     * 
+     * @param gvrContext
+     *            current {@link GVRContext}
+     * @param gvrActivity
+     *            a {@link GVRActivity}
+     * @param text
+     *            {@link String} to show on the textView
+     */
+
+    public GVRTextViewSceneObject(GVRContext gvrContext,
+            GVRActivity gvrActivity, String text) {
+        this(gvrContext, gvrActivity, DEFAULT_QUAD_WIDTH, DEFAULT_QUAD_HEIGHT,
+                text);
+    }
+
+    /**
+     * Shows a {@link TextView} on a {@linkplain GVRSceneObject scene object}
+     * with both view's default height and width and quad's default height and
+     * width. And the default text of this TextView will be "Hello World!"
+     * 
+     * @param gvrContext
+     *            current {@link GVRContext}
+     * @param gvrActivity
+     *            a {@link GVRActivity}
+     */
+    public GVRTextViewSceneObject(GVRContext gvrContext, GVRActivity gvrActivity) {
+        this(gvrContext, gvrActivity, "Hello World!");
```

Comment:
This seems wrong. Imho, the point of a constructor that doesn't take any initial text is that you will be setting it later. The default text, here, should probably be `""`.